hypothesis-python/src/hypothesis/searchstrategy/types.py

```diff
@@ -185,6 +187,10 @@ def from_typing_type(thing):
         typing.io.BinaryIO: st.builds(io.BytesIO, st.binary()),  # type: ignore
         typing.io.TextIO: st.builds(io.StringIO, st.text()),  # type: ignore
         typing.Reversible: st.lists(st.integers()),
+        typing.Collection: st.lists(st.integers()),
+        typing.Sequence: st.lists(st.integers()),
+        typing.Sized: st.lists(st.integers()),
+        typing.Iterable: st.lists(st.integers()),
         typing.SupportsAbs: st.complex_numbers(),
         typing.SupportsComplex: st.complex_numbers(),
         typing.SupportsFloat: st.complex_numbers(),
```

Comment:
Hmm, here's a bug!  (which I'm pretty sure *I* wrote, oops :disappointed_relieved:)  The correct mapping is 

```
SupportsFloat -> floats()    # not complex_numbers() - both
SupportsInt   -> floats()    # float(1j) and int(1j) are errors
```